### DISCUSSION TITLE
btConvexHullShape::project hides overloaded virtual function.

### DISCUSSION THREAD

**Discussion by stolk** [bulletphysics/bullet3#3469] [General]
When building with clang++, the compiler issues a warning on hiding an overloaded virtual function.

...
In file included from /home/bram/src/bullet3/src/btBulletCollisionCommon.h:32:
/home/bram/src/bullet3/src/BulletCollision/CollisionShapes/btConvexHullShape.h:76:15: warning: 'btConvexHullShape::project' hides overloaded virtual function [-Woverloaded-virtual]
        virtual void project(const btTransform& trans, const btVector3& dir, btScalar& minProj, btScalar& maxProj, btVector3& witnesPtMin,btVector3& witnesPtMax) const;
                     ^
/home/bram/src/bullet3/src/BulletCollision/CollisionShapes/btConvexShape.h:55:15: note: hidden overloaded virtual function 'btConvexShape::project' declared here: different number of parameters
      (4 vs 6)
        virtual void project(const btTransform& trans, const btVector3& dir, btScalar& min, btScalar& max) const;
                     ^
1 warning generated.

**Comment by erwincoumans** [comment]
That is likely a bug, although the code is not active by default (unless you enable the SAT clipping code, using 'convex->initializePolyhedralFeatures()' and world->getDispatchInfo()->m_enableSatConvex=true; )
Thanks for reporting, I'll fix it soon.

**Comment by erwincoumans** [comment]
(also I will be fixing some remaining bugs in btMultiBody and then it could be useful to create tanks/vehicles you might want to try that out later!)

**Comment by stolk** [comment]
Thanks Erwin.
Yes, at some point I may need some help with those tanks. The first port attempt from OpenDE to Bullet yielded a very unstable sim, even with miniscule timesteps, so I need to play around with different settings and solvers.
